### DISCUSSION TITLE
对于05年之前的数据，如何计算HEI2015

### DISCUSSION THREAD

**Discussion by love520lfh** [jamesjiadazhan/dietaryindex#249] [Q&A] [answered]
05年之前数据计算 HEI2015，能不能提供代码实例

**Comment by xysifang** [comment]
错误于load("NHANES_20052006.rda"): 
復原文件幻数出错(文件可能有损坏)-- 没有载入任何数据
此外: 警告信息:
file ‘NHANES_20052006.rda’ has magic number ''
  Use of save versions prior to 2 is deprecated 
不知大神用的哪个版本，读不进来，谢谢~

**Reply by jamesjiadazhan** [reply, marked as answer]
I don't have any problems opening the NHANES_20052006.rda file using R 4.3.2. 

Here I have 2 suggestions:
1. download the dietaryindex package again. when dietaryindex hints your upgrade the dependency, please upgrade all.
2. open a new R session

**Comment by jamesjiadazhan** [comment]
> 05年之前数据计算 HEI2015，能不能提供代码实例

Sorry this is a very late response to your request but here is the code:
- https://jamesjiadazhan.github.io/dietaryindex_manual/reference/HEI2015_NHANES_MPED.html

**Reply by zh-Mugen** [reply]
Sorry to bother you, but why does aMED have one less alcohol indicator than MEDI? The result is that aMED has a total score of 0-8, while MEDI has a score of 0-9.

**Reply by jamesjiadazhan** [reply]
MEDI is derived from the PREDIMED study. aMED is developed by a research team at Harvard. They are from different researchers. This is why,

**Reply by zh-Mugen** [reply]
According to for a given food component, if the intake is above the median, the participant scores 1. But why would aMED still have a score of 0.5? 
Link: https://view.officeapps.live.com/op/view.aspx?src=https%3A%2F%2Fraw.githubusercontent.com%2Fjamesjiadazhan%2Fdietaryindex%2Fmain%2Fdietaryindex_SCORING_ALGORITHM.xlsx&wdOrigin=BROWSELINK

**Reply by jamesjiadazhan** [reply]
This is a very late response. Sorry about that. I think the reason is because the results averaged the day 1 and day 2 aMED and one of them is 0 and another 1 is 1.

**Comment by 313983854** [comment]
The sample code of  setting up survey design for NHANES data in 2017-2018 does not fit the 1999-2000 one ,that is because I can not find out WTDRD1 and SDMVPSU in a same dataframe.
Plus, if I want to calculate HEI-2015 for 1999-2018, how to rectify the weight on your code?
Thank you so much for developing this pakage.

**Reply by 313983854** [reply]
I tried to cope with this with the following code: 

nutrient_data <- NHANES_19992000$NUTRIENT[, c("SEQN", "WTDRD1")]
demo_data <- NHANES_19992000$DEMO[, c("SEQN", "SDMVPSU", "SDMVSTRA")]
merged_data <- merge(nutrient_data, demo_data, by = "SEQN")
filtered_data <- merged_data[!is.na(merged_data$WTDRD1), ]
NHANES_19992000_design_d1 <- filtered_data[, c("SEQN", "SDMVPSU", "SDMVSTRA", "WTDRD1")]

Did I follow your rules?

**Reply by jamesjiadazhan** [reply]
Please refer to NHANES website for more information: 
https://wwwn.cdc.gov/nchs/nhanes/tutorials/weighting.aspx.
